src/Collection/Collection.service.ts

```diff
@@ -172,10 +173,16 @@ export class CollectionService {
     }
 
     try {
-      ethers.utils.verifyMessage(signedMessage, signature) // Throws if invalid
+      const address = ethers.utils.verifyMessage(signedMessage, signature) // Throws if invalid
+
+      if (signer.toLowerCase() !== address.toLowerCase()) {
```

Comment:
Maybe we should actually check the address belongs to the managers list of the collection?

Reply:
We can, but the contract is already doing that on its own.